### DISCUSSION TITLE
New Discord Username Format

### DISCUSSION THREAD

**Discussion by ghost** [discord-net/Discord.Net#2678] [General]
Per https://discord.com/blog/usernames

Are there going to be any breaking changes from this switch?

Usernames will go from 

`Username#1234 -> @Username`

I have a property like this
```csharp
public string DiscordUsername => $"{Context.User.Username}#{Context.User.Discriminator}";
```
I am guessing as users switch over, Discriminator will just switch to null and should be nbd? I only use this as an informational piece, it won't break any of my code.

**Comment by exsersewo** [comment]
In  the [how to prepare](https://support-dev.discord.com/hc/en-us/articles/13667755828631#h_01GYA8Q7TRFF2Y16B1F9XGCK31) section of the dev article, it'll just send back 0 instead of null.

Depending on how the end user implements user filtering, the discriminator being 0 is the only "breaking" change.

As Username would just go to the new Username, and `DisplayName` would be a new property needed to be added for Users (on both Discord and D.Nets side).

**Reply by ghost** [reply]
> In the [how to prepare](https://support-dev.discord.com/hc/en-us/articles/13667755828631#h_01GYA8Q7TRFF2Y16B1F9XGCK31) section of the dev article, it'll just send back 0 instead of null.
> 
> Depending on how the end user implements user filtering, the discriminator being 0 is the only "breaking" change.
> 
> As Username would just go to the new Username, and `DisplayName` would be a new property needed to be added for Users (on both Discord and D.Nets side).

Okay, I mostly just use this as an informational property, and it shouldn't break any of my code. I can probably just leave it as-is, and wait for the new update and switch to the `DisplayName` property.

The majority of my mission critical code uses the user's `Id`, so not super worried.

Glad to hear there won't be anymore breaking changes 😄

**Reply by markski1** [reply]
I'd be all for the DisplayName method to be extended to the main user object, that way it can be used in functions that expect any IUser instead of only for GuildUsers
